### DISCUSSION TITLE
Will it work on a AMD chip ?

### DISCUSSION THREAD

**Discussion by RajdeepVerma** [Qonfused/OSX-Hyper-V#11] [Q&A] [answered]
Will it work on a AMD chip ?
I have Ryzen 5 4600H, i have seen macos running tricking hardly work with AMD cpu ?

**Comment by Qonfused** [comment, marked as answer]
Yes, this works on AMD CPUs as well. You'll want to check out https://github.com/Qonfused/OSX-Hyper-V/issues/5 and use the AMD kernel patch linked in the post. It's explained in the Dortania guide as well -- you'll want to follow the same process for patching AMD CPUs.

**Reply by RajdeepVerma** [reply]
Thank you so much for your quick response.

**Comment by Ryan-kopp** [comment]
I am trying to get this to run on Server 2025 with Hyper-V on an AMD Ryzen CPU. I am a little confused on Step 5 and how to modify the  config.Plist and add the patches.plist

![caputre1](https://github.com/user-attachments/assets/1910f567-6a35-407d-a147-4d45d24a39ea)
![Capture2](https://github.com/user-attachments/assets/e8f7ca28-b52f-4722-9d28-f0af568bf875)

**Reply by Qonfused** [reply]
You'll want to use a plist editor like [ProperTree](github.com/corpnewt/ProperTree) to copy the entries over to the config.plist. Essentially the [AMD kernel patches](https://github.com/AMD-OSX/AMD_Vanilla/blob/master/patches.plist) contain the `Kernel -> Patch` entries you'll need to move over to the config.plist after building/downloading.

The only manual steps after that is adjusting the four core-count patches (`algrey - Force cpuid_cores_per_package`) as mentioned in the AMD Vanilla README based on the number of CPU cores you're allocating to the VM and adding the wrmsr patch (also provided in a similar plist).

I've also add updated the README to include instructions for generating these patches using the [`amd.py`](https://github.com/Qonfused/OCE-Build/blob/main/scripts/amd.py) script from the OCE-Build repository.

**Reply by Ryan-kopp** [reply]
Thank you @Qonfused , can you take a look at my plist file I think I got it edited correctly,
[config - Copy.txt](https://github.com/user-attachments/files/20676572/config.-.Copy.txt)
 but I am still at the same place

**Reply by Qonfused** [reply]
Your patches look correct. If you can get an OpenCore log with these debug options it should show where you're getting stuck in the boot process:

https://github.com/Qonfused/OSX-Hyper-V/blob/2532f6d3e8fa3dc8ebdc78d5d3a7e4fd7284a011/src/config.yml#L79-L102

These logs should output as .txt files at the root of the EFI partition (next to the EFI folder). Mainly we're interested in the properties for kernel debugging (which includes the boot args).

**Reply by Ryan-kopp** [reply]
I only saw one error I am not really sure how to debug or analyze the log file
[opencore-2025-06-10-204444.txt](https://github.com/user-attachments/files/20683787/opencore-2025-06-10-204444.txt)

**Reply by Qonfused** [reply]
The OpenCore log contains EDK2 mneumonics which are in general hard to decipher, but in this case the relevant line is this one:
```
105:573 00:579 AAPL: #[EB.MM.AKM|!] Err(0xE) <- EB.MM.MKP
```

You're facing the same error as #44 where too little memory is allocated to the VM. Any macOS version since Big Sur in my experience will require at least 6-8 GB of memory (at least for the installer).

I did also notice that two kernel patches aren't being applied, though the patches themselves look correct.
```
04:905 102:768 OC: Kernel patcher result 9 for kernel (algrey | _cpuid_set_generic_info | Remove wrmsr(0x8B) | 10.13+) - Not Found
104:993 00:088 OC: Kernel patcher result 10 for kernel (algrey | _cpuid_set_generic_info | Replace rdmsr(0x8B) with constant 186 | 10.13+) - Not Found
```

For now I'd retry with a higher amount of RAM and try to get a screenshot/video of where it gets stuck (assuming it gets further into the installer).

**Reply by Ryan-kopp** [reply]
Thanks @Qonfused, I meant to bump up the VM memory to 8GB but forgot to go back on update it.  I updated it and as you said it got further in the boot process. Attached is new log file and screen shot where it gets stuck.  It seems to hang there should it continue and exit out into a reboot?

[opencore-2025-06-11-063317.txt](https://github.com/user-attachments/files/20692109/opencore-2025-06-11-063317.txt)
![image](https://github.com/user-attachments/assets/0c835d51-4a9c-46fe-80f6-df5a8937f15f)

**Reply by Qonfused** [reply]
It should continue after that point.

The `#[EB|LOG:EXITBS:START]` line means OpenCore has finally handed off to the installer (so subsequent logs on screen will be from the macOS/installer boot.efi). It's also worth pointing out that anything further won't be displayed in the log file (since we're no longer in OpenCore), though you can record/screenshot any points afterwards.

I'm mainly worried about these two lines, which seems to indicate some kind of error with runtime relocation of firmware code/data lines during the boot transition (from OpenCore into the installer).
```
#[EB.RH.LRH|P!] Err(0x3) <- EB.LD.LR
#[EB.BST.FBS|!] Err(0x3) <- EB.RH.LRH
```
The main thing that comes to mind is secure boot (double check this is disabled in VM settings), though it could still be caused from the two MSR patches failing to apply. In any case, what is causing this is not entirely clear.

Assuming you've applied the debugging config.plist settings from https://github.com/Qonfused/OSX-Hyper-V/discussions/11#discussioncomment-13425882, do you get anything further (not present in the OpenCore log) or does it immediately reboot/hang?

**Reply by Ryan-kopp** [reply]
![{AFD66C6C-981E-4FBD-BAFA-9385F7763CD7}](https://github.com/user-attachments/assets/02db4d7b-53dd-4d31-b146-b2534fb037f7)

**Reply by Ryan-kopp** [reply]
It just hangs at that screen, is there anything missing from here? 

![image](https://github.com/user-attachments/assets/64d2dc1e-854d-4d0b-b7f0-231f5b30bfc4)
